### DISCUSSION TITLE
Use python to precict energe and force  appear NAN

### DISCUSSION THREAD

**Discussion by lammps-vasp** [deepmodeling/deepmd-kit#957] [Q&A]
Hello developers
when using Python to predict the energy and force of the system, the following errors occur,Is there a problem with the file format of POSCAR.
I look forward to your reply

OMP: Info #252: KMP_AFFINITY: pid 195016 tid 195281 thread 32 bound to OS proc set 4
WARNING: Radial neighbor list length of type WARNING: Radial neighbor list length of type 0 is not enough
WARNING: Radial neighbor list length of type WARNING: Radial neighbor list length of type 0 is not enough
0 is not enough
0 is not enough
WARNING: Radial neighbor list length of type 0WARNING: Radial neighbor list length of type  is not enough 0 is not enough

The following is the force department data. Why does Nan appear？
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
-1.197523057147747519e-05 -1.054037052199862900e-05 -1.218992308461597194e-05
1.943206335093612402e-01 -1.369947061264624455e-01 -1.477661677741639812e+00
-1.522996864567707576e+01 4.846521819483145421e+00 -3.672573223496319628e+01
0.000000000000000000e+00 0.000000000000000000e+00 0.000000000000000000e+00
-2.302142409933168599e-01 2.235872094221092787e+00 2.461147891180890834e+00
-2.460351848454363499e-01 -2.399223129852091560e-02 -7.000173273169632004e-01
1.801148076476546400e+00 -1.758314252960498481e+00 8.384101045342062797e-01
1.396825812560401007e-05 -1.001951570965637565e-05 -8.935249966104508412e-06
3.575854634112127228e-02 -1.023718862372784491e-02 2.348444406788349204e-01
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
nan nan nan
1.127616585878146171e-01 2.257022413167124597e+00 4.449787150460618790e+00

**Comment by njzjz** [comment]
Can you provide the model and coordinates? Also, note that `WARNING: Radial neighbor list length of type 0 is not enough` means `sel` is not enough.

**Reply by lammps-vasp** [reply]
After adjusting sel again just now, it will still prompt the previous error. I think it may be the problem of POSCAR, because the coordinates of POSCAR are extracted from OUTCAR. May it be some format problems caused by extraction? The following are POSCAR, model and predicted energy and force.


Very much looking forward to the answer

**Reply by njzjz** [reply]
I don't see your attachment. Also, could you check if some of your atoms have the same coordinates? This is the most likely reason to produce NaN (not a number).

**Comment by lammps-vasp** [comment]
After adjusting sel again just now, it will still prompt the previous error. I think it may be the problem of POSCAR, because the coordinates of POSCAR are extracted from OUTCAR. May it be some format problems caused by extraction? The following are POSCAR, model and predicted energy and force.


Very much looking forward to the answer




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "deepmodeling/deepmd-kit"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年8月13日(星期五) 凌晨1:31
***@***.***&gt;;
抄送:&nbsp;"Zhifeng ***@***.******@***.***&gt;;
主题:&nbsp;Re: [deepmodeling/deepmd-kit] Use python to precict energe and force  appear NAN (#957)





 
Can you provide the model and coordinates? Also, note that WARNING: Radial neighbor list length of type 0 is not enough means sel is not enough.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.


从QQ邮箱发来的超大附件

frozen_model.pb (8.02M, 无限期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?k=22336565b8f04a9e243e8c3d1631074e57575606060702524b575c56051c01595f02485d0805004c055652000953000304555556303e3507145c1f005e6e580e0256094b4053355c&t=exs_ftn_download&code=f3ee015a




force0811.raw (236.3K, 无限期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?k=77663864a67acdc9776bd13c4234031901525b56070d5000185109015c1905550452155d5752051b05545e525107035201000b57643931505a145b01540c00071b1459136409&t=exs_ftn_download&code=5f8dd416




energe0811.raw (4B, 无限期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?k=7b3631629fcdc8c77a3bd83a453205170f0054030050040b15010806021f035901541c0305050e155b5300075107515a08550251633c375d5653430506020f090918430314320a&t=exs_ftn_download&code=861bc278




POSCAR (550.5K, 无限期)进入下载页面：http://mail.qq.com/cgi-bin/ftnExs_download?k=5d326130e554c09d243f88681e31054d525405030a5304034b5607040b1c035052024c090d04064f00515004000154575500520238373732296122716a310a&t=exs_ftn_download&code=f2a0817b
